File: app/src/main/java/com/example/ugcssample/drone/UsbAccessoryManager.java
package com.example.ugcssample.drone;

import android.hardware.usb.UsbAccessory;

public interface UsbAccessoryManager {

    interface Listener {
        /**
         * There is a USB accessor, which is ready for use.
         */
        void onUsbAccessoryConnected(UsbAccessory usbAccessory);

        /**
         * No available USB assessor.
         */
        void onUsbAccessoryDisconnected();
    }

    void init();

    void check();

}

File: app/src/main/java/com/example/ugcssample/drone/UsbAccessoryManager.kt
package com.example.ugcssample.drone

import android.hardware.usb.UsbAccessory

interface UsbAccessoryManager {
    interface Listener {
        /**
         * There is a USB accessor, which is ready for use.
         */
        open fun onUsbAccessoryConnected(usbAccessory: UsbAccessory?)

        /**
         * No available USB assessor.
         */
        open fun onUsbAccessoryDisconnected()
    }

    open fun init()
    open fun check()
}